indicator-stickynotes.py

```diff
@@ -256,8 +256,13 @@ def handler(indicator):
     #indicator.nset.notes[current_note].show()
     #indicator.nset.current_note = current_note
 
+def reload_handler(indicator):
+    # reload from data file on SIGUSR2
+    with open(os.path.expanduser(indicator.data_file), encoding="utf-8") as fsock:
+        indicator.nset.merge(fsock.read())
+    install_glib_handler(indicator, signal.SIGUSR2)
```

Comment:
I guess the app was written with the assumption that the user is the only one making notes so there is no concurrency management in there. Scripts making notes with quick repetition will simply overwrite each other's changes - as all the application data is written in one massive json file right now.

A solution to this could be implementation of proper concurrency management by locking the config file before writing it using an appropriate os-call. That would effectively serialize changes. This could also mean waiting a lot of time in case there are a lot of changes to be written - not a typical scenario though. Also, different notes could be written in different files which would also help.